source-code/website/src/pages/install/InstallationProvider.tsx

```diff
@@ -0,0 +1,261 @@
+import { type JSXElement, createEffect } from "solid-js"
+import { openRepository, createNodeishMemoryFs } from "@lix-js/client"
+import { publicEnv } from "@inlang/env-variables"
+import {
+	LocalStorageProvider,
+	getLocalStorage,
+	useLocalStorage,
+} from "#src/services/local-storage/index.js"
+import { InlangConfig, tryCatch } from "@inlang/app"
+import type { Step } from "./index.page.jsx"
+import { marketplaceItems } from "@inlang/marketplace"
+import type { RecentProjectType } from "#src/services/local-storage/src/schema.js"
+
+export function InstallationProvider(props: {
+	repo: string
+	modules: string[]
+	step: () => Step
+	setStep: (step: Step) => void
+	optIn: Record<string, any>
+	children: JSXElement
+}) {
+	const [localStorage, setLocalStorage] = useLocalStorage() ?? []
+	const user = localStorage?.user
+
+	/**
+	 * This function checks for common errors before repo initialization (to be more performant) and sets the step accordingly.
+	 */
```

Comment:
@floriandwt if this functions checks for common errors **before** a repo initialization, why is this reactive und not a `validateRepo` function that is called once?

Reply:
Shouldn't run once because the user opts in to the process and the app has to proceed based on the input